ansible-scylla-node/tasks/ssl.yml

```diff
@@ -24,37 +37,70 @@
         issuer:
           commonName: "{{ scylla_cluster_name }}.internal"
         provider: selfsigned
+      when: ca_crt_details.stat.exists == False
   delegate_to: localhost
   run_once: true
 
 - name: Generate keys signed by the local CA and push them to the nodes
   block:
+    - name: Check hostname folder exists
+      stat:
+        path: "./ssl/{{ item }}"
+      with_inventory_hostnames:
+        - scylla
+      register: hostname_folder_details
+      delegate_to: localhost
+
     - name: Create a directory for the key
       file:
-        path: "./ssl/{{ inventory_hostname }}"
+        path: "./ssl/{{ item }}"
```

Comment:
Why this change was needed? What's "item"?

Reply:
On line 59 I have added:

```yaml
      with_items: "{{ hostname_folder_details.results }}"
      when: item.stat.isdir == False
```

This creates the item and means we can use that when creating the folder. It is required as this deals with the list of hostnames from the inventory. Maybe I'm doing this the wrong way? But I didn't see another way to check if all the folders existed and create them if they didn't...

Reply:
Then, I wonder why we still need `inventory_hostname` in the namespace - it seems that you have eliminated all its usages but it still seems to be defined:
 
```
      with_inventory_hostnames:
        - scylla
```

@dyasny @tarzanek Please, comment.

Reply:
it's defined by ansible, you cannot remove inventory_hostname...

Reply:
Maybe I did it in the wrong way, I just found that this method worked in my case, but there might be a more graceful way to do this?

Reply:
`inventory_hostname` and `with_inventory_hostnames` are builtins, pointing to the hostnames ansible gets from the inventory. 

`inventory_hostname` is the current host that is being processed by ansible
`with_inventory_hostnames` is a generator for a list of these hostnames, so we can iterate over them. 

when iterating, the current processed list member is referred to as `item`

Reply:
Actually thinking back to Friday this does explain why I saw ansible do something undesirable...

It is already be looping over the hosts, then I have created a second loop inside that loop as for 3 hosts, it printed out 9 lines in my terminal, sorry, I forgot I hadn't fixed that...

I'm not sure how to fix it right now, but it shouldn't loop in the loop like that...

Reply:
I am confused, so how should it loop? 
what should be looped over what?
let's write it here in plain english and translate to ansible python after we all agree what it should do

Reply:
At the moment, the ansible script loops over the hostnames from the inventory, but when I tried to detect whether the files/folders for each hostname existed and only create the files if they didn't exist I hit issues. I found this solution worked, but I had to add another loop inside, which probably isn't required, but it works, it just runs the operation more times than desired. If there is a way to check for the existence of the folders, the private key and the public part without having the additional loop created by lines 49/50 then it would be ideal.

Reply:
I removed this part as it wasn't needed, sorry :)